### DISCUSSION TITLE
Allocate memory for each thread

### DISCUSSION THREAD

**Discussion by Tonaion02** [bevyengine/bevy#16040] [Q&A]
I am iterating on parallel on some entities, i need some buffers to save some information.
For now i'm allocating and de-allocating memory each time. But i see that is something really stupid to do.
I don't need to re-allocate each time this buffer.
I need only to have a different buffer of memory for each thread and make a clear each time on it.
Is it possible to do something like that in Bevy?

**Comment by Arrekin** [comment]
You can use thread local variables(each thread will get its own value to work with). I am using them for my pathfinding alogrithms and they work very well. https://doc.rust-lang.org/std/macro.thread_local.html (it's a general Rust feature)

**Reply by Tonaion02** [reply]
Probably is the best solution for my problem, because i don't need at a certain point to make an access to all the buffers from one thread. For example if i need to collect for each thread some data, and then collect all the data, can become less easier to do that.
In that case probably we can achieve a solution with: https://bevy-cheatbook.github.io/programming/par-iter.html#scoped-tasks.

**Reply by Arrekin** [reply]
It really depends on what you are trying to achieve. The link I posted is good in situations when your work in threads is independent and you want to use these variables in many different functions.
If your goal is to have one system that spawns tasks, computes them and then returns something then `par_iter()` could be a good solution. You can use it in addition to `Local<>` resources that could allow you to persist your allocated memory across frames: https://bevy-cheatbook.github.io/programming/local.html
